procedural/utils/src/main/java/gov/nasa/ammos/aerie/procedural/utils/TypeUtilsEditablePlanAdapter.java

```diff
@@ -0,0 +1,185 @@
+import gov.nasa.jpl.aerie.merlin.driver.MissionModel;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import gov.nasa.ammos.aerie.procedural.scheduling.plan.EditablePlan;
+import gov.nasa.ammos.aerie.procedural.scheduling.simulation.SimulateOptions;
+import gov.nasa.jpl.aerie.merlin.protocol.types.SerializedValue;
+import gov.nasa.jpl.aerie.orchestration.simulation.SimulationUtility;
+import gov.nasa.jpl.aerie.scheduler.plan.MerlinToProcedureSimulationResultsAdapter;
+import gov.nasa.ammos.aerie.procedural.timeline.payloads.activities.AnyDirective;
+import gov.nasa.ammos.aerie.procedural.timeline.payloads.activities.Directive;
+import gov.nasa.ammos.aerie.procedural.timeline.payloads.activities.DirectiveStart;
+import gov.nasa.jpl.aerie.types.ActivityDirective;
+import gov.nasa.jpl.aerie.types.ActivityDirectiveId;
+import gov.nasa.jpl.aerie.types.Plan;
+import kotlin.jvm.functions.Function1;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * An {@link EditablePlan} implementation for the type-utils {@link Plan} class.
+ * Allows for local simulation and plan editing, such as for custom scheduler drivers,
+ * constraint checkers, or other workflows that don't use the Aerie database.
+ *
+ * To use, create a {@link SimulationUtility} and use that to load the mission model.
+ * Then create a {@link Plan} object; you can create one with the constructor or by using
+ * {@link gov.nasa.jpl.aerie.orchestration.PlanJsonParser} to load a plan from file.
+ * Then you can create an editable plan object like this:
+ *
+ * ```
+ * editablePlan = new DefaultEditablePlanDriver(
+ *     new TypeUtilsEditablePlanAdapter(
+ *         new TypeUtilsPlanAdapter(myPlan),
+ *         simUtility,
+ *         model
+ *     )
+ * );
+ * ```
+ *
+ * You can then edit and simulate the plan using the {@link EditablePlan} interface. You can pass it
+ * to a scheduling goal, a constraint, or anything else that needs the plan and sim results. The official
+ * constraint interface takes an immutable plan and sim results separately, so you can call a constraint like
+ * this: `new MyConstraint(...).run(plan, plan.simulate())`. (The {@link EditablePlan} will be upcast to
+ * the immutable {@link gov.nasa.ammos.aerie.procedural.timeline.plan.Plan} interface.
+ */
+public class TypeUtilsEditablePlanAdapter implements gov.nasa.ammos.aerie.procedural.timeline.plan.Plan,
+    DefaultEditablePlanDriver.PlanEditAdapter
+{
+
+  private final TypeUtilsPlanAdapter plan;
+  private final SimulationUtility simUtility;
+  private final MissionModel<?> model;
+  private long idCounter;
+  private SimulationResults latestResults;
+
+  private boolean changedSinceLastSim = true;
+
+  public TypeUtilsEditablePlanAdapter(
+      TypeUtilsPlanAdapter plan,
+      SimulationUtility simUtility,
```

Comment:
Major: I get the following error from this class: `Cannot resolve symbol "SimulationUtility". Add dependency on module "orchestration-utils"`. Please update the `build.gradle` to resolve this.

Reply:
The `build.gradle` already does have a dependency on `orchestration-utils`. Try a clean build?

Reply:
For some reason `clean`ing wasn't effective and I had to hard-refresh my dependencies, but yes, the issue has gone away, and I see the entry in `build.gradle`